### PR TITLE
fix bug #1396

### PR DESCRIPTION
**Pull request by shaomeng**
The root of this issue is that `Throwing exceptions from an event handler is not supported in Qt.`

I changed it so that an error of our custom type is reported. Please review and determine if this is a good solution.

### CODE REVIEW

File: apps/vaporgui/VaporTable.h [hunk 1]
```diff
@@ -165,17 +167,24 @@ public slots:
...
+    template<typename T>
+    operator T() const   //implicitly convert into T
+    {
+        std::stringstream ss(_value);
+        T convertedValue;
+        if ( ss >> convertedValue )
+            return convertedValue;
+        else
+        {
+            std::string msg = "conversion failed: " + _value;
```
Comment (clyne) [inline apps/vaporgui/VaporTable.h]: This won't fly either. If <T> is a string the else clause will return a NULL. Moreover there is no error checking performed for by VaporTable::GetValue() - the only user of Value - so the error message recorded by SetErrMsg will never be reported.

Not sure what should be done here.

**Comment by shaomeng** [comment]
I would think a more reasonable design is to have ”VaporTable" truthfully
return a string that's stored there, and each individual widget whoever
makes use of "VaporTable" to take the responsibility of converting a string
to the data type it prefers...

**Comment by shaomeng** [comment]
Since the issue requires better fix than this band-aid, the issue is kicked back to @sgpearse and this PR is closed.

**Comment by shaomeng** [comment]
This isn't even a band-aid. VAPOR still crashes occasionally with this fix. There's a deeper problem...

**Pull request closed without merging**